gpytorch/lazy/added_diag_lazy_tensor.py

```diff
@@ -125,3 +129,21 @@ def _init_cache_for_non_constant_diag(self, eye, batch_shape, n):
         logdet = self._r_cache.diagonal(dim1=-1, dim2=-2).abs().log().sum(-1).mul(2)
         logdet -= (1.0 / self._noise).log().sum([-1, -2])
         self._precond_logdet_cache = logdet.view(*batch_shape) if len(batch_shape) else logdet.squeeze()
+
+    @cached(name="svd")
+    def _svd(self) -> Tuple["LazyTensor", Tensor, "LazyTensor"]:
+        diag = self._diag_tensor.diag()
+        if torch.equal(diag, diag[..., :1].expand(diag.shape)):
+            U, S_, V = self._lazy_tensor.svd()
+            S = S_ + diag  # this assumes all diagonal entries are positive
+            return U, S, V
+        return super()._svd()
+
+    @cached(name="symeig")
+    def _symeig(self, eigenvectors: bool = False) -> Tuple[Tensor, Optional[LazyTensor]]:
+        diag = self._diag_tensor.diag()
+        if torch.equal(diag, diag[..., :1].expand(diag.shape)) and torch.all(diag[..., :1] >= 0):
+            evals_, evecs = self._lazy_tensor.symeig(eigenvectors=eigenvectors)
+            evals = evals_ + diag  # this assumes all diagonal entries are positive
+            return evals, evecs
```

Comment:
As in the other comment: We probably don't want to include these because the backwards call of the symeig of the matrix plus the diagonal doesn't necessarily match the result of this operation, due to gradient un-identifiability. This should be what's causing the test failures.
```suggestion
```

Reply:
The better fix is to create an `AddedConstDiag` lazy tensor and update the unit tests to better match that.